validation/src/test/scala/hmda/validation/engine/ts/validity/TsValidityEngineSpec.scala

```diff
@@ -0,0 +1,50 @@
+package hmda.validation.engine.ts.validity
+
+import hmda.model.fi.ts.{ Contact, Respondent }
+import hmda.parser.fi.ts.TsGenerators
+import hmda.validation.rules.ts.validity.ValidityUtils
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{ MustMatchers, PropSpec }
+import org.scalatest.prop.PropertyChecks
+
+import scala.concurrent.ExecutionContext
+
+class TsValidityEngineSpec extends PropSpec with PropertyChecks with MustMatchers with TsGenerators with TsValidityEngine with ScalaFutures with ValidityUtils {
+  override implicit val ec: ExecutionContext = scala.concurrent.ExecutionContext.Implicits.global
+
+  property("Transmittal Sheet must be valid") {
+    forAll(tsGen) { ts =>
+      whenever(respondentNotEmpty(ts.respondent)) {
+        validate(ts).isSuccess mustBe true
+      }
+    }
+  }
+
+  property("Transmittal Sheet fails V105 (Respondent Mailing Address)") {
```

Comment:
I believe this line and line 41 (the one naming a test that claims to be about V155) are switched.